src/contracts/middleware/StakeRegistry.sol

```diff
@@ -0,0 +1,481 @@
+     * @dev Precondition:
+     *          1) `quorumBitmaps[i]` should be the bitmap that represents the quorums that `operators[i]` registered for
+     * @dev reverts if there are no operators registered with index out of bounds
+     */
+    function updateStakes(address[] calldata operators, bytes32[] calldata operatorIds, uint192[] calldata quorumBitmaps, uint256[] calldata prevElements) external {
+        // for each quorum, loop through operators and see if they are apart of the quorum
+        // if they are, get their new weight and update their individual stake history and thes
+        // quorum's total stake history accordingly
+        for (uint8 quorumNumber = 0; quorumNumber < quorumCount;) {
+            OperatorStakeUpdate memory totalStakeUpdate;
+            // for each operator
+            for(uint i = 0; i < operatorIds.length;) {
+                // if the operator is apart of the quorum
+                if (quorumBitmaps[i] >> quorumNumber & 1 == 1) {
+                    // if the total stake has not been loaded yet, load it
+                    if (totalStakeUpdate.updateBlockNumber == 0) {
+                        totalStakeUpdate = _totalStakeHistory[quorumNumber][_totalStakeHistory[quorumNumber].length - 1];
+                    }
+                    bytes32 operatorId = operatorIds[i];
+                    // update the operator's stake based on current state
+                    (uint96 stakeBeforeUpdate, uint96 stakeAfterUpdate) = _updateOperatorStake(operators[i], operatorId, quorumNumber);
+                    // calculate the new total stake for the quorum
+                    totalStakeUpdate.stake = totalStakeUpdate.stake - stakeBeforeUpdate + stakeAfterUpdate;
+                }
+                unchecked {
+                    ++i;
+                }
+            }
+            // if the total stake for this quorum was updated, record it in storage
+            if (totalStakeUpdate.updateBlockNumber != 0) {
+                // update the total stake history for the quorum
+                _recordTotalStakeUpdate(quorumNumber, totalStakeUpdate);
+            }
+            unchecked {
+                ++quorumNumber;
+            }
+        }
+
+        // record stake updates in the EigenLayer Slasher
+        for (uint i = 0; i < operators.length;) {
+            serviceManager.recordStakeUpdate(operators[i], uint32(block.number), serviceManager.latestServeUntilBlock(), prevElements[i]);
+            unchecked {
+                ++i;
+            }
+        }     
+    }
+
+    // INTERNAL FUNCTIONS
+
+    function _registerOperator(address operator, bytes32 operatorId, bytes memory quorumNumbers) internal {
+        uint8 quorumNumbersLength = uint8(quorumNumbers.length);
+        // check the operator is registering for only valid quorums
+        require(uint8(quorumNumbers[quorumNumbersLength - 1]) < quorumCount, "StakeRegistry._registerStake: greatest quorumNumber must be less than quorumCount");
```

Comment:
typo: `_registerStake` => `_registerOperator`

Reply:
this is also repeated elsewhere

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/0aa1b58a86c05f0e2888020267abee81618d46d7